Valigator.Core/Data.cs

```diff
@@ -117,5 +119,65 @@ private static IDataContainer<TValue> GetErrorDataContainer(IDataContainer<TValu
 
 		public static implicit operator TValue(Data<TValue> data)
 			=> data.Value;
+
+		public override bool Equals(object obj) 
+			=> obj is Data<TValue> data 
+				&& this == data;
+
+		public static bool operator ==(Data<TValue> x, Data<TValue> y)
+			=> x.DataDescriptor.Equals(y.DataDescriptor)
+					&& (x._value.Equals(y._value)
+						|| (IsEnumerable<TValue>() && IsSequenceEqual(x._value as IEnumerable, y._value as IEnumerable)));
+
+
+		private static bool IsSequenceEqual(IEnumerable a, IEnumerable b)
+		{
+				var firstEnumerator = a.GetEnumerator();
+				var secondEnumerator = b.GetEnumerator();
+
+			while(firstEnumerator.MoveNext() && secondEnumerator.MoveNext())
+			{
+				if (!(IsEnumerable(firstEnumerator.Current) && IsEnumerable(secondEnumerator.Current) ?
+					IsSequenceEqual(firstEnumerator.Current as IEnumerable, secondEnumerator.Current as IEnumerable)
+					: AreObjectsEqual(firstEnumerator.Current, secondEnumerator.Current)))
+					return false;
+			}
+				return firstEnumerator.MoveNext() == secondEnumerator.MoveNext();	
```

Comment:
Arrays with exactly one more element than the other will incorrectly return true here.